src/main/scala/mesosphere/mesos/ResourceUtil.scala

```diff
@@ -130,12 +159,12 @@ object ResourceUtil {
   def consumeResources(
     resources: Iterable[MesosProtos.Resource],
     usedResources: Iterable[MesosProtos.Resource]): Iterable[MesosProtos.Resource] = {
-    val usedResourceMap: Map[ResourceMatchKey, Seq[MesosProtos.Resource]] =
-      usedResources.groupBy(ResourceMatchKey(_)).mapValues(_.to[Seq])
+    val usedResourceMap: Map[ResourceConsumeKey, Seq[MesosProtos.Resource]] =
+      usedResources.groupBy(ResourceConsumeKey(_)).mapValues(_.to[Seq])
 
     resources.flatMap { resource: MesosProtos.Resource =>
-      usedResourceMap.get(ResourceMatchKey(resource)) match {
-        case Some(usedResources: Seq[MesosProtos.Resource]) =>
+      usedResourceMap.get(ResourceConsumeKey(resource)) match {
+        case Some(usedResources: Iterable[MesosProtos.Resource]) =>
```

Comment:
why Iterable vs. immutable Seq? we're trying to avoid Iterable and standardize on immutable Seq